### PR TITLE
change encryption from rsa to age (X25519)

### LINKED ISSUE #3: research other encryption options instead of rsa
RSA has unfortunate limits on message size (446 bytes for keypair of 4096 bit size), which prevent us from encoding public key in one message. Another option is to pack multiple messages (ciphertexts) in one packet at encoding.Encoder side, but this will complicate packet structure a lot.

Some options:
- https://pkg.go.dev/filippo.io/age#pkg-examples
- https://github.com/ethereum/go-ethereum/blob/master/crypto/ecies/ecies.go
- https://github.com/google/tink/blob/master/docs/GOLANG-HOWTO.md

### PR DESCRIPTION
**Pull request by pymq**
closes #3

### CODE REVIEW

File: encoding/encoding.go [hunk 1]
```diff
@@ -100,54 +89,32 @@ func (e *Encoder) UnpackMessage(encodedBody []byte) ([]byte, MessageType, error)
...
-func GenerateKey() (*rsa.PrivateKey, *rsa.PublicKey, error) {
-	private, err := rsa.GenerateKey(rand.Reader, rsaKeySize)
+func GenerateKey() (*age.X25519Identity, error) {
+	private, err := age.GenerateX25519Identity()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
-	return private, &private.PublicKey, nil
+	return private, nil
```
Comment (GrigoryKrasnochub) [inline encoding/encoding.go]: You can retrun result of "age.GenerateX25519Identity". There is no custom error text, so i don't see any reason for processing error here.

File: encoding/encoding.go [hunk 2]
```diff
@@ -100,54 +89,32 @@ func (e *Encoder) UnpackMessage(encodedBody []byte) ([]byte, MessageType, error)
...
 
 	flags := binary.BigEndian.Uint64(decoded[:8])
-	message, err := rsa.DecryptOAEP(
-		sha256.New(),
-		rand.Reader,
-		e.ownPrivKey,
-		decoded[8:],
-		rsaLabel,
-	)
+	r, err := age.Decrypt(bytes.NewReader(decoded[8:]), e.ownPrivKey)
```
Comment (GrigoryKrasnochub) [inline encoding/encoding.go:95]: I'm afraid of out of array. I'm not sure that there is any problem, but check could simplify debbuging for us in future.

**Review by GrigoryKrasnochub — APPROVED** [review]
Looks nice for me. You did a great job!

File: encoding/encoding.go (hunk 2 above)
Comment (pymq) [reply encoding/encoding.go:95]: Good catch, fixed

**Pull request merged**